### PR TITLE
Bugfix wrong handling of payments case when all tasks are settled

### PR DESCRIPTION
**Pull request by dybi**
resolves: https://github.com/golemfactory/concent/issues/1091

### CODE REVIEW

File: concent_api/core/message_handlers.py [hunk 1]
```diff
@@ -823,40 +817,44 @@ def handle_send_force_payment(
...
+            reason=message.concents.ForcePaymentRejected.REASON.NoUnsettledTasksFound,
         )
+    provider_force_payment_commited = message.concents.ForcePaymentCommitted(
+        payment_ts              = parse_datetime_to_timestamp(claim_against_requestor.closure_time),
+        task_owner_key          = requestor_ethereum_public_key,
+        provider_eth_account    = provider_eth_address,
+        amount_paid             = claim_against_requestor.amount_as_int,
+        amount_pending          = 0,
+        recipient_type          = message.concents.ForcePaymentCommitted.Actor.Provider,
+    )
```
Comment (kbeker) [inline concent_api/core/message_handlers.py]: Ou man, style :)

File: concent_api/core/message_handlers.py [hunk 2]
```diff
@@ -823,40 +817,44 @@ def handle_send_force_payment(
...
-            recipient_type          = message.concents.ForcePaymentCommitted.Actor.Requestor,
-        )
+    requestor_force_payment_commited = message.concents.ForcePaymentCommitted(
+        payment_ts              = parse_datetime_to_timestamp(claim_against_requestor.closure_time),
+        task_owner_key          = requestor_ethereum_public_key,
+        provider_eth_account    = provider_eth_address,
+        amount_paid             = claim_against_requestor.amount_as_int,
+        amount_pending          = 0,
+        recipient_type          = message.concents.ForcePaymentCommitted.Actor.Requestor,
+    )
```
Comment (kbeker) [inline concent_api/core/message_handlers.py]: Same here :-)

File: concent_api/core/tests/test_integration_force_payment.py [hunk 3]
```diff
@@ -275,7 +278,77 @@ def test_provider_send_force_payment_with_no_value_to_be_paid_concent_should_rej
         with freeze_time("2018-02-05 12:00:20"):
             with mock.patch(
                 'core.message_handlers.bankster.settle_overdue_acceptances',
-                return_value=None
+                side_effect=BanksterNoUnsettledTasksError()
+            ) as settle_overdue_acceptances:
+                response = self.send_request(
+                    url='core:send',
+                    data                                = serialized_force_payment,
+                )
```
Comment (kbeker) [inline concent_api/core/tests/test_integration_force_payment.py]: And here

File: concent_api/core/tests/test_integration_force_payment.py [hunk 4]
```diff
@@ -275,7 +278,77 @@ def test_provider_send_force_payment_with_no_value_to_be_paid_concent_should_rej
...
+        )
+
+        self._test_response(
+            response,
+            status       = 200,
+            key          = self.PROVIDER_PRIVATE_KEY,
+            message_type = message.concents.ForcePaymentRejected,
+            fields       = {
+                'reason':    message.concents.ForcePaymentRejected.REASON.NoUnsettledTasksFound,
+                'timestamp': parse_iso_date_to_timestamp("2018-02-05 12:00:20"),
```
Comment (kbeker) [inline concent_api/core/tests/test_integration_force_payment.py]: And here

File: concent_api/core/payments/bankster.py [hunk 5]
```diff
@@ -138,7 +140,7 @@ def claim_deposit(
             # we can't add a new claim.
             if provider_deposit <= aggregated_claims_against_provider['sum_of_existing_claims'] + settings.ADDITIONAL_VERIFICATION_COST:
                 claim_against_requestor.delete()
-                raise TooSmallProviderDeposit
+                raise BanksterTooSmallProviderDepositError()
```
Comment (rwrzesien) [inline concent_api/core/payments/bankster.py]: Maybe add error message here too.

File: concent_api/core/message_handlers.py [hunk 6]
```diff
@@ -791,14 +793,6 @@ def handle_send_force_payment(
     for subtask_results_accepted in client_message.subtask_results_accepted_list:
         validate_task_to_compute(subtask_results_accepted.task_to_compute)
 
-    if sum([subtask_results_accepted.task_to_compute.price
```
Comment (rwrzesien) [inline concent_api/core/message_handlers.py]: Why is this removed?

File: concent_api/core/tests/test_payments_bankster.py [hunk 7]
```diff
@@ -343,7 +345,7 @@ def assert_mocked_sci_functions_were_called(self, get_list_of_payments_call_coun
         self.validate_list_of_transaction_mock.assert_called_once()
         self.assertEqual(self.get_list_of_payments_mock.call_count, get_list_of_payments_call_count)
 
-    def test_that_settle_overdue_acceptances_should_return_none_if_subtask_costs_where_already_paid(self):
+    def test_that_settle_overdue_acceptances_should_rasie_no_unsettled_tasks_exception_when_all_tasks_are_paid_off(self):
```
Comment (rwrzesien) [inline concent_api/core/tests/test_payments_bankster.py]: `rasie` => `raise`

File: concent_api/core/tests/test_payments_bankster.py [hunk 8]
```diff
@@ -356,27 +358,29 @@ def test_that_settle_overdue_acceptances_should_return_none_if_subtask_costs_whe
...
 
-        claim_against_requestor = self.call_settle_overdue_acceptances_with_mocked_sci_functions()
+        with self.assertRaises(BanksterNoUnsettledTasksError):
+            self.call_settle_overdue_acceptances_with_mocked_sci_functions()
 
-        self.assertIsNone(claim_against_requestor)
         self.assert_mocked_sci_functions_were_called()
 
-    def test_that_settle_overdue_acceptances_should_return_none_if_requestor_deposit_value_is_zero(self):
+    def test_that_settle_overdue_acceptances_should_raise_too_small_requestor_deposit_exception_when_requestor_has_insufficient_funds(self):
```
Comment (rwrzesien) [inline concent_api/core/tests/test_payments_bankster.py]: `rasie` =>`raise`

File: concent_api/core/tests/test_payments_bankster.py [hunk 9]
```diff
@@ -356,27 +358,29 @@ def test_that_settle_overdue_acceptances_should_return_none_if_subtask_costs_whe
...
         """
         In this test we have following calculations:
 
-        TaskToCompute price:                                    3000 (does not matter in this case)
+        TaskToCompute price:                                    12000 (does not matter in this case)
```
Comment (rwrzesien) [inline concent_api/core/tests/test_payments_bankster.py]: Why this changed? Text says that it should not matter.

File: concent_api/core/payments/bankster.py (hunk 5 above)
Comment (dybi) [reply concent_api/core/payments/bankster.py]: ok

File: concent_api/core/tests/test_payments_bankster.py (hunk 8 above)
Comment (dybi) [reply concent_api/core/tests/test_payments_bankster.py]: ? :) it is **raise**

File: concent_api/core/tests/test_payments_bankster.py (hunk 9 above)
Comment (dybi) [reply concent_api/core/tests/test_payments_bankster.py]: ok, I will refactor it to get better clarity ;)
